src/lib/es2015.iterable.d.ts

```diff
@@ -8,11 +8,18 @@ interface SymbolConstructor {
     readonly iterator: symbol;
 }
 
-interface IteratorResult<T> {
-    done: boolean;
-    value: T;
+interface IteratorYieldResult<TYield> {
+    done: false;
+    value: TYield;
 }
 
+interface IteratorReturnResult<TReturn> {
+    done: true;
+    value: TReturn;
+}
+
+type IteratorResult<T, TReturn = T | void> = IteratorYieldResult<T> | IteratorReturnResult<TReturn>;
+
 interface Iterator<T> {

```

Comment:
Why is this not changed? Does this PR not improve the experience when writing iterators manually (without generator syntax)?